src/course-partner/course-partner.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import _ from 'lodash';
+import { PartnerService } from 'src/partner/partner.service';
+import { CoursePartnerRepository } from './course-partner.repository';
+
+@Injectable()
+export class CoursePartnerService {
+  constructor(
+    @InjectRepository(CoursePartnerRepository)
+    private coursePartnerRepository: CoursePartnerRepository,
+    private readonly partnerService: PartnerService,
+  ) {}
+
+  async getCoursePartnersByCourseId(courseId: string) {
+    return await this.coursePartnerRepository
+      .createQueryBuilder('course_partner')
+      .leftJoinAndSelect('course_partner.partner', 'partner')
+      .where('course_partner.courseId = :courseId', { courseId })
+      .getMany();
+  }
+
+  async updateCoursePartners(partners: string[], courseId: string) {
+    const coursePartners = await this.getCoursePartnersByCourseId(courseId);
+
+    const partnersObjects = await Promise.all(
+      partners.map(async (partner) => {
+        if (partner === 'public' || partner === 'Public')
+          return await this.partnerService.getPartner('Chayn');
+        return await this.partnerService.getPartner(partner);
+      }),
+    );
+
+    //If course existed in included_for_partners but now doesnt
+    Promise.all(
+      coursePartners.map(async (cp) => {
+        if (!_.find(partnersObjects, { id: cp.partner.id })) {
+          cp.active = false;
+          await this.coursePartnerRepository.save(cp);
+        }
+      }),
+    );
+
+    return Promise.all(
+      partnersObjects.map(async (partner) => {
+        const coursePartner = await this.coursePartnerRepository.findOne({
```

Comment:
we already have _all_ the coursePartners from line 24, so can we just do a `coursePartners.find(...` here to get the existing cousePartner object, instead of calling the db another x times?

Reply:
What this function is doing is it's going through the existing course partner data and seeing which one doesnt exist in the partner objects that's being sent from storyblok. If it doesnt exist then set active to false. If the id doesnt exist in the include_for_partner array then set active to false.

Reply:
yep gotcha but this comment is specifically about we're recalling the `this.coursePartnerRepository.findOne({` to get each partner, but we already have the data for all the existing course partners, when we call `const coursePartners = await this.getCoursePartnersByCourseId(courseId);` on line 24

so why do we need to call each one individually when we already have the list?